keras/layers/convolutional.py

```diff
@@ -1247,6 +1250,189 @@ def get_config(self):
         return dict(list(base_config.items()) + list(config.items()))
 
 
+class Deconvolution3D(Convolution3D):
+    '''Transposed convolution operator for filtering windows of three-dimensional inputs.
+    The need for transposed convolutions generally arises from the desire
+    to use a transformation going in the opposite direction of a normal convolution,
+    i.e., from something that has the shape of the output of some convolution
+    to something that has the shape of its input
+    while maintaining a connectivity pattern that is compatible with said convolution. [1]
+
+    When using this layer as the first layer in a model,
+    provide the keyword argument `input_shape`
+    (tuple of integers, does not include the sample axis),
+    e.g. `input_shape=(3, 128, 128, 128)` for a 128x128x128 voxel volume with three channels.
+
+    To pass the correct `output_shape` to this layer,
+    one could use a test model to predict and observe the actual output shape.
+
+    # Examples
+
+    ```python
+        # apply a 3x3x3 transposed convolution with stride 1x1x1 and 3 output filters on a 12x12x12 volume:
+        model = Sequential()
+        model.add(Deconvolution3D(3, 3, 3, 3, output_shape=(None, 3, 14, 14, 14), border_mode='valid', input_shape=(3, 12, 12, 12)))
+        # Note that you will have to change the output_shape depending on the backend used.
+
+        # we can predict with the model and print the shape of the array.
+        dummy_input = np.ones((32, 3, 12, 12, 12))
+        # For TensorFlow dummy_input = np.ones((32, 12, 12, 12, 3))
+        preds = model.predict(dummy_input)
+        print(preds.shape)
+        # Theano GPU: (None, 3, 13, 13, 13)
+        # Theano CPU: (None, 3, 14, 14, 14)
+        # TensorFlow: (None, 14, 14, 14, 3)
```

Comment:
These examples are confusing. dim ordering should be referred to as "th dim ordering / tf dim ordering". Please pick one of them and have the entire example use the same convention. You may then follow up with the same example in the alternative dim ordering.